packages/block-editor/src/components/rich-text/index.native.js

```diff
@@ -445,6 +448,27 @@ function RichTextWrapper(
 				isURL( plainText.trim() )
 			) {
 				mode = 'BLOCKS';
+				// Embed handler
+				embedHandlerPickerRef.current?.presentPicker( {
+					createEmbed: () => {
+						const content = pasteHandler( {
```

Comment:
I was wondering if since this pasteHandler functionality is similar to [this logic](https://github.com/WordPress/gutenberg/pull/35204/files#diff-1bf4290b2ff8a9f2d347d3450576c98613f26f26f4ce05b850cf232483f7b28cL450) it should be extracted to a function. Personally, I was fine with leaving it as is, as it makes the intent clear and showcases what is generating the content. I wouldn't mind a second opinion though.

Reply:
Overall, the code within `onPaste` handler could be extracted to smaller functions to make it more readable. For this case, looks like we're introducing a bit of redundancy but I'm fine keeping the duplicated code.

One thing that would be interesting to try is to extract the whole `onPaste` to a helper, something similar as we have in the web version with the [`usePasteHandler`](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/rich-text/use-paste-handler.js), I wouldn't like to delay the PR due to this comment so I'd consider this only if we have spare time to test it, wdty?

Reply:
> Overall, the code within onPaste handler could be extracted to smaller functions to make it more readable. For this case, looks like we're introducing a bit of redundancy but I'm fine keeping the duplicated code.

Understood!

> One thing that would be interesting to try is to extract the whole onPaste to a helper, something similar as we have in the web version with the usePasteHandler, I wouldn't like to delay the PR due to this comment so I'd consider this only if we have spare time to test it, wdty?

Makes sense! The logic is quite extensive and would be better served in a helper. I will see how best this can be done 👍🏾